core/distributed/index_map_kernels.hpp

```diff
@@ -72,11 +76,36 @@ namespace kernels {
         experimental::distributed::index_space is, array<_ltype>& local_ids)
 
 
+/**
+ * This kernels maps local indices to global indices.
+ *
+ * The relevant input parameter from the index map are:
+ *
+ * - partition: the global partition
+ * - remote_global_idxs: the remote global indices, segmented by the owning part
+ *                       ids
+ * - rank: the part id of this process
+ *
+ * Any local index that is not part of the specified index space is mapped to
+ * invalid_index.
+ */
```

Comment:
I think this is a nice way to capture the documentation of our kernels (And we have a lot of them). I wonder if it is possible to list all the kernels and capture their documentation in the developer documentation.

Reply:
I agree with Marcel's comment yesterday that Ginkgo developers would rather use code as documentation than the Doxygen, I think we should focus on writing good comments, regardless of whether they work precisely as Doxygen comments.